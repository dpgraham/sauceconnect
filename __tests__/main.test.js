jest.mock('child_process')
const main = require('../');
const childProcess = require('child_process');
const { EventEmitter } = require('events');

describe('main', function () {
	let exitSpy, mockSpawnEventEmitter;
	beforeAll(function () {
		childProcess.spawn.mockImplementation(() => {
			mockSpawnEventEmitter = new EventEmitter();
			return mockSpawnEventEmitter;
		});
		exitSpy = jest.spyOn(process, 'exit');
		exitSpy.mockImplementation(() => {});
	});
	test('should run the executable', async function () {
		const bin = {
			run: jest.fn().mockReturnValue(Promise.resolve(true)),
			path: jest.fn().mockReturnValue('/bin/sauceconnect')
		}
		await main(bin, ['bar', '--foo'])
		expect(bin.run).toBeCalledTimes(1)
		expect(childProcess.spawn).toBeCalledWith('/bin/sauceconnect', ['bar', '--foo'], expect.any(Object))
		mockSpawnEventEmitter.emit('exit', 0);
		expect(exitSpy.mock.calls).toEqual([[0]]);
	});
	describe('.binWrapper', function () {
		test('should create a binary wrapper', async function () {
			const bw = await main.binWrapper();
			expect(Array.isArray(bw._src)).toEqual(true);
			expect(typeof bw._dest).toEqual('string');
		});
		test('should respect the SAUCECONNECT_INSTALL_BINARY wrapper', async function () {
			const bw = await main.binWrapper('http://some-fake-url');
			expect(bw._src).toMatchSnapshot();
			expect(typeof bw._dest).toEqual('string');
		});

		test('should respect the SAUCECONNECT_INSTALL_BINARY_MIRROR wrapper', async function () {
			const bw = await main.binWrapper(null, 'https://some-fake-mirror-url');
			expect(Array.isArray(bw._src)).toEqual(true);
			expect(bw._src[0].url).toMatch(/^https:\/\/some-fake-mirror-url\//);
			expect(typeof bw._dest).toEqual('string');
		});
	});
});
