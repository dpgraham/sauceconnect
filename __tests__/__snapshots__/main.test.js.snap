// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main .binWrapper should respect the SAUCECONNECT_INSTALL_BINARY wrapper 1`] = `
Array [
  Object {
    "arch": "",
    "os": "",
    "url": "http://some-fake-url",
  },
]
`;
